### ISSUE TITLE
Pass RouteSettings through Dialogs

### ISSUE THREAD

**Issue by davidlantrip** [flutter/flutter#31261]
It looks like `RouteSettings` aren't passed through the `showDialog` methods. Being able to attach them to dialogs would be useful for adding metadata about the current route.

```
Future<T> showGeneralDialog<T>({
  @required BuildContext context,
  @required RoutePageBuilder pageBuilder,
  bool barrierDismissible,
  String barrierLabel,
  Color barrierColor,
  Duration transitionDuration,
  RouteTransitionsBuilder transitionBuilder,
}) {
  assert(pageBuilder != null);
  assert(!barrierDismissible || barrierLabel != null);
  return Navigator.of(context, rootNavigator: true).push<T>(_DialogRoute<T>(
    pageBuilder: pageBuilder,
    barrierDismissible: barrierDismissible,
    barrierLabel: barrierLabel,
    barrierColor: barrierColor,
    transitionDuration: transitionDuration,
    transitionBuilder: transitionBuilder,
  ));
}
```

It looks like adding them to that method and passing them into the `push` call would work?  Then having `showDialog` /  `showCupertinoDialog`  pass them through might be sufficient?
Something similar could be done for `showCupertinoModalPopup`.

**Comment by hpoul** [comment]
fyi, i have created a simple PR for this: https://github.com/flutter/flutter/pull/38631 let's hope somebody looks at it :-)

**Comment by darrenaustin** [comment]
Should be fixed with #51525. Thanks @hpoul!
